miden-lib/asm/miden/note.masm

```diff
@@ -129,3 +129,110 @@ export.get_sender
     syscall.get_note_sender
     # => [sender]
 end
+
+#! Computes hash of note inputs starting at the specified memory address.
+#!
+#! Inputs:  [inputs_ptr, num_inputs]
+#! Outputs: [HASH]
+#!
+#! Panics if num_inputs is greater than 128.
+export.compute_inputs_hash
+    swap
+    # => [num_inputs, inputs_ptr]
+
+    # check that number of inputs is less than 128
+    dup push.128 u32lte assert
```

Comment:
There is no guarantees that `num_inputs` is a `u32` value - so, this should be:
```
dup push.128 u32assert2 u32lte assert
```
Or we could use `u32assert2` earlier to make sure both `inputs_ptr` and `num_inputs` are `u32` values.

Also, would be good to add error codes to asserts.

Reply:
Recently, I have had one thought: in case we are not doing the `u32assert2` check and the memory address will exceed the `u32::MAX`, during the `mem_stream` we will get the specific error message: `"Memory address cannot exceed 2^32 but was {addr}"`. But if we do the `u32assert2` check we will get just a generic `"Not a u32 value"` message. So is it really better to have this `u32assert2` check? Both errors will be an Execution Errors, so just an error message will be different.

Reply:
Yes, if `mem_stream` gets called before we transform the address in any way, it is better to leave bound-checking to `mem_stream`. But we still need to use `u32assert2` for checking of the `num_inputs`.